tools/lan_choose_character_kart/lan_choose_character_kart.py

```diff
@@ -0,0 +1,299 @@
+                             global_symbols: list[str]) -> None:
+    with open(filename, 'w', encoding='utf-8') as symbolsfile:
+        equ_symbol_lines = '\n'.join([f'.equ {sym}, {equ_symbols[sym]}'
+                                      for sym in equ_symbols]) + '\n'
+        global_symbol_lines = '\n'.join([f'.global {adr}' for adr in global_symbols]) + '\n'
+        symbolsfile.write(equ_symbol_lines + global_symbol_lines)
+
+
+def prepare_symbols_ld_file(temp_dir: tempfile.TemporaryDirectory, region: str,
+                            start_adr: int) -> None:
+    symbolsld_path = os.path.join(temp_dir, 'symbols.ld')
+
+    funcsymbol_defs = '\n'.join([
+        f"{k} = {symbols.asmfuncsymbols[region][k]} - {f'0x{start_adr:08x}'};"
+        for k in symbols.asmfuncsymbols[region]
+    ])
+    with open(symbolsld_path, 'w', encoding='utf-8') as symbol_ldfile:
+        symbol_ldfile.write(f'''\
+SECTIONS
+{{
+    . = {f"0x{start_adr}"};
+    .text : {{
+        {funcsymbol_defs}
+    }}
+}}
+''')
+
+
+@dataclass
+class CodeBlockInfo:
+    curasmfile: str
+    blockstartname: str
+    startoffset: int
+    addressestoresolve: tuple[str]
+    funcaddressestoresolve: tuple[str]
+    tablerowdata: list
+
+
+def make_data_file(region: str, temp_dir: tempfile.TemporaryDirectory, asm_dir: str) -> bytes:
+    symbolsinc_path = os.path.join(temp_dir, 'symbols.inc')
+    symbolsld_path = os.path.join(temp_dir, 'symbols.ld')
+
+    codeblocks = (CodeBlockInfo(
+        curasmfile="patchlogoapp_part2.s",
+        blockstartname="NewCalcCodeBlockStart",
+        startoffset=symbols.symbols[region]["OsakoM____sinit_ResMgr_cpp"],
+        addressestoresolve=("NetGateAppAfterCt", "NewDrawCodeBlockStart",
+                            "SetRandSeedAfterLanEntryCt"),
+        funcaddressestoresolve=("LANEntryCalcPrintMenu", "ResolveConsoleAndControllerIDs",
+                                "PrepareKartInfo", "CheckIfAllKartsHaveFinished"),
+        tablerowdata=[]),
+                  CodeBlockInfo(
+                      curasmfile="lanentrydraw_beforeseconddraw.s",
+                      blockstartname=None,
+                      startoffset=symbols.symbols[region]["DriverDataChild__mDriverDataDefault"],
+                      addressestoresolve=("LANEntryDrawBeforeSecondDraw", ),
+                      funcaddressestoresolve=(),
+                      tablerowdata=[]))
+    for cb in codeblocks:
+        prepare_symbols_inc_file(symbolsinc_path, symbols.symbols[region], cb.addressestoresolve)
+        curasmfilepath = os.path.join(asm_dir, cb.curasmfile)
+        tempobj_path = os.path.join(temp_dir, 'temp.o')
+
+        run_subprocess([
+            devkit_tools.ASPATH, curasmfilepath, '-mregnames', '-I', temp_dir, '-I', asm_dir, '-o',
+            tempobj_path
+        ])
+        objdumpoutput = run_subprocess([devkit_tools.OBJDUMPPATH, '-t',
+                                        tempobj_path]).stdout.decode("utf-8")
```

Comment:
As per the other suggestion in the `run_subprocess()` function, this would just be:

```suggestion
        objdumpoutput = run_subprocess([devkit_tools.OBJDUMPPATH, '-t', tempobj_path])
```